test/Microsoft.Sbom.Api.Tests/SbomValidatorTests.cs

```diff
@@ -0,0 +1,225 @@
+using System.Threading.Tasks;
+using Microsoft.Sbom.Api.Entities;
+using Microsoft.Sbom.Api.Output.Telemetry;
+using Microsoft.Sbom.Api.Workflows;
+using Microsoft.Sbom.Common;
+using Microsoft.Sbom.Common.Config;
+using Microsoft.Sbom.Common.Config.Validators;
+using Microsoft.Sbom.Contracts;
+using Microsoft.Sbom.Extensions;
+using Microsoft.Sbom.Extensions.Entities;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Constants = Microsoft.Sbom.Api.Utils.Constants;
+
+namespace Microsoft.Sbom.Api.Tests;
+
+[TestClass]
+public class SbomValidatorTests
+{
+    private Mock<IWorkflow<SbomParserBasedValidationWorkflow>> workflowMock;
+    private Mock<IRecorder> recorderMock;
+    private Mock<IEnumerable<ConfigValidator>> configValidatorsMock;
+    private Mock<IConfiguration> configurationMock;
+    private Mock<ISbomConfigProvider> sbomConfigProviderMock;
+    private Mock<IFileSystemUtils> fileSystemUtilsMock;
+    private Mock<ISbomConfig> sbomConfigMock;
+
+    // Common test data
+    private readonly string buildDropPath = "/test/drop";
+    private readonly string outputPathFile = "/test/output.json";
+    private readonly string outputPathDirectory = "/test/output";
+    private readonly List<SbomSpecification> specifications = new List<SbomSpecification> { new SbomSpecification("SPDX", "2.2") };
+    private readonly string manifestDirPath = "/test/manifest";
+    private readonly ManifestInfo manifestInfo = Constants.TestManifestInfo;
+    private readonly string manifestJsonPath = "/test/manifest/manifest.json";
+
+    [TestInitialize]
+    public void Init()
+    {
+        workflowMock = new Mock<IWorkflow<SbomParserBasedValidationWorkflow>>(MockBehavior.Strict);
+        recorderMock = new Mock<IRecorder>(MockBehavior.Strict);
+        configValidatorsMock = new Mock<IEnumerable<ConfigValidator>>(MockBehavior.Strict);
+        configurationMock = new Mock<IConfiguration>(MockBehavior.Strict);
+        sbomConfigProviderMock = new Mock<ISbomConfigProvider>(MockBehavior.Strict);
+        fileSystemUtilsMock = new Mock<IFileSystemUtils>(MockBehavior.Strict);
+        sbomConfigMock = new Mock<ISbomConfig>(MockBehavior.Strict);
+    }
+
+    [TestCleanup]
+    public void AfterEachTest()
+    {
+        workflowMock.VerifyAll();
+        recorderMock.VerifyAll();
+        configValidatorsMock.VerifyAll();
+        configurationMock.VerifyAll();
+        sbomConfigProviderMock.VerifyAll();
+        fileSystemUtilsMock.VerifyAll();
+        sbomConfigMock.VerifyAll();
+    }
+
+    [TestMethod]
+    public async Task ValidateSbomAsync_WithNoErrorsAndNoExceptions_ReturnsTrue()
+    {
+        var errors = new List<FileValidationResult>();
+        var exceptions = new List<Exception>();
+
+        configValidatorsMock.Setup(cv => cv.GetEnumerator()).Returns(new List<ConfigValidator>().GetEnumerator());
+
+        configurationMock.Setup(c => c.ManifestInfo).Returns(new ConfigurationSetting<IList<ManifestInfo>>
```

Comment:
How much of this boilerplate setup code can we reasonably pull into a helper method? A quick glance suggests that most of lines 72-87 are shared, other than the contents of `errors` and `exceptions`, which are easy to pass in as parameters to the helper